packages/mongo/mongoAsyncUtils.js

```diff
@@ -0,0 +1,15 @@
+// Returns an async function that will be executed at most one time,
+// no matter how often you call it. Useful for lazy initialization.
+export const onceAsync = func => {
+  let ran = false;
+  let memo = undefined;
+  return async function executeOnce() {
+    if (ran) return memo;
+    const memoPromise = func.apply(this, arguments);
+    memo = await memoPromise;
+    func = null;
+    ran = true;
+    return memo;
+  };
+};
```

Comment:
Here is an alternative:

```js
export const onceAsync = (thunk) => {
  let memo;
  return async function executeOnce() {
      if (thunk /* maybe use type checking ? */) {
        const memoPromise = Reflect.apply(thunk, this, arguments);
        memo = await memoPromise;
        thunk = null;
      }
      return memo;
   };
};
```

One could also use a class to reduce the footprint of the scope bound function `executeOnce`.

Reply:
This is obvious bike-shedding. Feel free to ignore!